File: examples/check7.js
let invoice;

const __rubyStrftime = (format) => {
  const date = new Date();
  const replacements = {
    "%Y": String(date.getFullYear()),
    "%m": String(date.getMonth() + 1).padStart(2, "0"),
    "%d": String(date.getDate()).padStart(2, "0")
  };
  return String(format ?? "").replace(/%[Ymd]/g, (match) => replacements[match] ?? match);
};

class Invoice {
  constructor(client, amount) {
    this.__client = client;
    this.__amount = amount;
  }
  to_text() {
    return `======================
     FACTURE
======================
Client : ${this.__client}
Montant: ${this.__amount} €
Date   : ${__rubyStrftime("%d/%m/%Y")}
======================
`;
  }
}
invoice = new Invoice("Alice Dupont", 250);
console.log(invoice.to_text());

File: examples/Old_test/check7.rb
class Invoice
  def initialize(client, amount)
    @client = client
    @amount = amount
  end

  def to_text
    <<~INVOICE
      ======================
           FACTURE
      ======================
      Client : #{@client}
      Montant: #{@amount} €
      Date   : #{Time.now.strftime("%d/%m/%Y")}
      ======================
    INVOICE
  end
end

# --- Utilisation ---
invoice = Invoice.new("Alice Dupont", 250)
puts invoice.to_text